### PR TITLE
Picolibc

### PR DESCRIPTION
**Pull request by keith-packard**
picolibc support - requires updated freedom-metal and freedom-devicetree-tools

### CODE REVIEW

File: scripts/libmetal.mk [hunk 1]
```diff
@@ -108,6 +108,9 @@ $(METAL_LIB_DIR)/stamp: $(METAL_BUILD_DIR)/Makefile
 
 $(METAL_LIB_DIR)/lib%.a: $(METAL_LIB_DIR)/stamp ;@:
 
+$(METAL_LIB_DIR)/libmetal-pico.a: $(METAL_LIB_DIR)/libriscv__pico__metal.a
```
Comment (nategraff-sifive) [inline scripts/libmetal.mk]: If we name the library what we want in freedom-metal, we can avoid defining a rule to rename it.

File: scripts/standalone.mk [hunk 2]
```diff
@@ -32,6 +32,18 @@ ifeq ($(RISCV_CMODEL),)
...
+SPEC=nano
+endif
+
+ifeq ($(SPEC),nano)
+LIBMETAL_EXTRA=-lmetal-gloss
+endif
+
+ifeq ($(SPEC),picolibc)
+LIBMETAL_EXTRA=-lmetal-pico
+endif
```
Comment (nategraff-sifive) [inline scripts/standalone.mk]: Using SPEC internally works, but I'd prefer to have a more descriptive option on the frontend for users to actually choose their libc implementation like `RISCV_LIBC` or similar

File: scripts/libmetal.mk (hunk 1 above)
Comment (keith-packard) [reply scripts/libmetal.mk]: yeah, I'm not sure why I had this funky name; I'll try to switch it.

File: scripts/standalone.mk (hunk 2 above)
Comment (keith-packard) [reply scripts/standalone.mk]: Or just LIBC? And add some error checking so if the value is set to something crazy, it exits instead of breaks.

Comment (keith-packard) [reply scripts/standalone.mk]: And I added error checking so you get a message from make when you use an invalid libc value. I haven't added any documentation about the supported values yet.

**Comment by keith-packard** [comment]
Ok, this branch has been updated to use Nate's BSP generation script.

File: scripts/standalone.mk [hunk 3]
```diff
@@ -36,6 +36,24 @@ ifeq ($(RISCV_CMODEL),)
 RISCV_CMODEL = medany
 endif
 
+ifeq ($(RISCV_LIBC),)
```
Comment (bsousi5) [inline scripts/standalone.mk]: I like this. Thanks for doing this. @knightsifive has been asking for this.

**Comment by keith-packard** [comment]
> Given that pico lib is not available as part of the current/standard toolchain bundle, can you add a write up in README.md to where one can go about download and use it.

Hrm. Maybe I could publish a freedom-tools add-on for picolibc until freedom-tools provides a release that includes picolibc?

**Comment by nategraff-sifive** [comment]
Always building libmetal-pico.a breaks when using newlib-nano, so we should only build libmetal-pico.a when we need it. Just pushed two commits to bump freedom metal to enable and use the extra configure option.

**Comment by nategraff-sifive** [comment]
DNM yet, waiting for submodule merges.

**Pull request merged**